sm-core/src/main/java/com/salesmanager/core/business/modules/cms/category/CategoryFileManagerImpl.java

```diff
@@ -0,0 +1,135 @@
+package com.salesmanager.core.business.modules.cms.category;
+
+import com.salesmanager.core.business.constants.Constants;
+import com.salesmanager.core.business.exception.ServiceException;
+import com.salesmanager.core.business.modules.cms.product.ProductFileManager;
+import com.salesmanager.core.business.modules.cms.product.ProductImageGet;
+import com.salesmanager.core.business.modules.cms.product.ProductImagePut;
+import com.salesmanager.core.business.modules.cms.product.ProductImageRemove;
+import com.salesmanager.core.business.utils.CoreConfiguration;
+import com.salesmanager.core.business.utils.ProductImageCropUtils;
+import com.salesmanager.core.business.utils.ProductImageSizeUtils;
+import com.salesmanager.core.model.catalog.category.image.CategoryImage;
+import com.salesmanager.core.model.catalog.product.Product;
+import com.salesmanager.core.model.catalog.product.file.ProductImageSize;
+import com.salesmanager.core.model.catalog.product.image.ProductImage;
+import com.salesmanager.core.model.content.FileContentType;
+import com.salesmanager.core.model.content.ImageContentFile;
+import com.salesmanager.core.model.content.OutputContentFile;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.net.FileNameMap;
+import java.net.URLConnection;
+import java.util.List;
+
+/**
+ * @author Mostafa Saied <mostafa.saied.fci@gmail.com>
+ */
+
+public class CategoryFileManagerImpl extends CategoryFileManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CategoryFileManagerImpl.class);
+
+
+  private CategoryImagePut uploadImage;
+  private CategoryImageGet getImage;
+  private CategoryImageRemove removeImage;
+
+  public CategoryImageRemove getRemoveImage() {
+    return removeImage;
+  }
+
+  public CategoryImagePut getUploadImage() {
+    return uploadImage;
+  }
+
+  public void setUploadImage(CategoryImagePut uploadImage) {
+    this.uploadImage = uploadImage;
+  }
+
+  public CategoryImageGet getGetImage() {
+    return getImage;
+  }
+
+  public void setGetImage(CategoryImageGet getImage) {
+    this.getImage = getImage;
+  }
+
+  public void setRemoveImage(CategoryImageRemove removeImage) {
+    this.removeImage = removeImage;
+  }
+
+
+  public void addCategoryImage(CategoryImage categoryImage, ImageContentFile contentImage)
```

Comment:
what exactly implemented here? and why there are many streams opened!

Reply:
Actually me too don't understand all of it :)
We can think about it together if you want?